authentication/auth.go

```diff
@@ -0,0 +1,162 @@
+package authentication
+
+import (
+	"fmt"
+	"log"
+	"time"
+	"errors"
+	"regexp"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+	"net/url"
+
+	"github.com/satori/go.uuid"
+
+	"gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+
+	"github.com/collisdigital/go-launch-a-survey/settings"
+)
+
+func loadEncryptionKey() (key *rsa.PublicKey, error error) {
+
+	encryptionKeyPath := settings.GetSetting("JWT_ENCRYPTION_KEY_PATH")
+	if keyData, err := ioutil.ReadFile(encryptionKeyPath); err == nil {
+		block, _ := pem.Decode(keyData)
+		pub, err := x509.ParsePKIXPublicKey(block.Bytes)
+		if err == nil {
+			return pub.(*rsa.PublicKey), nil
```

Comment:
this is ok if you can guarantee the type, otherwise you can test it as part of the assertion, e.g.

```
if v, ok := pub.(*rsa.PublicKey); ok { // wont panic if you have the wrong type
    // v is no a *rsa.PublicKey
    return v, nil
}
```